src/components/Button.js

```diff
@@ -0,0 +1,27 @@
+import propTypes, { func } from 'prop-types';
+
+const Button = ({
+  iconOnly,
+  size,
+  icon: IconComponent,
+  children,
+  ...props
+}) => (
+  <button
+    className={
```

Comment:
suggestion: you can use [`classnames`](https://github.com/JedWatson/classnames) package for better classNames management.

Then you can define your classNames like this:

```jsx
import cx from 'classnames';

// ...

const iconClass = iconOnly && IconComponent ? 'icon' : undefined;
const sizeClass = size ? `button--${size}` : undefined;

return (
    <button
        classNames={cx('button', sizeClass, iconClass)}
        ...
    >...</button>
);
```